scripts/batched-llava-caption.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2022 MosaicML Diffusion authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Script to LLaVA caption an image dataset."""
+
+import os
+import threading
+import time
+from argparse import ArgumentParser, Namespace
+from typing import Optional
+
+import torch
+import wandb
+from composer.utils import dist
+from huggingface_hub import snapshot_download
+from torchvision import transforms
+from tqdm.auto import tqdm
+
+try:
+    from llava.constants import DEFAULT_IMAGE_TOKEN  # type: ignore
+    from llava.constants import DEFAULT_IM_END_TOKEN, DEFAULT_IM_START_TOKEN, IMAGE_TOKEN_INDEX  # type: ignore
+    from llava.conversation import SeparatorStyle, conv_templates  # type: ignore
+    from llava.mm_utils import get_model_name_from_path, tokenizer_image_token  # type: ignore
+    from llava.model.builder import load_pretrained_model  # type: ignore
+    from llava.utils import disable_torch_init  # type: ignore
+except ImportError:
+    raise ImportError(
+        'LLaVA is not installed. Please install it with `pip install llava@git+https://github.com/haotian-liu/LLaVA.git`'
+    )
+
+from PIL import Image, ImageOps
+from streaming import Stream
+from streaming.base import MDSWriter
+
+from diffusion.datasets.image import StreamingImageDataset
+
+
+class LLaVACaptioner:
+    """LLaVA captioner class."""
+
+    def __init__(self,
+                 model_name: str = 'liuhaotian/llava-v1.6-vicuna-13b',
+                 max_tokens: int = 1024,
+                 compile: bool = False,
+                 quantize: bool = False,
+                 multi_gpu: bool = False,
+                 device: Optional[torch.device] = None):
+        self.model_name = model_name
+        self.conv_mode = 'llava_v1'
+        self.max_tokens = max_tokens
+        self.device = torch.device('cuda') if device is None else device
+
+        self.tokenizer, self.model, self.image_processor, self.context_len = self.load_llava(quantize=quantize,
+                                                                                             multi_gpu=multi_gpu)
+        self.generate = self.model.generate
+        if compile:
```

Comment:
This api doesn't allow us to pass custom dict with custom compile options

Reply:
True, was trying to balance simplicity with flexibility here, but open to a suggestion to allow a custom config here.